packages/components/Input/src/Input.types.ts

```diff
@@ -0,0 +1,247 @@
+   */
+  spacingAssistiveTextStart?: number;
+
+  /**
+   * The label's start margin. Used to align it with input box.
+   */
+  spacingLabelStart?: number;
+
+  /**
+   * The label's top margin.
+   */
+  spacingLabelTop?: number;
+}
+
+export interface InputTokens extends InputCoreTokens {
+  /**
+   * States that can be applied to an input.
+   */
+  hasIcon?: InputTokens;
+  error?: InputTokens;
+  focused?: InputTokens;
+  typing?: InputTokens;
+  filled?: InputTokens;
+}
+
+export interface InputProps extends PressableFocusProps {
+  /*
+   * Source URL or name of the icon to show on the input.
+   */
+  icon?: IconSourcesType;
+
+  /**
+   * The width of the button.
+   */
+  width?: ViewStyle['width'];
+
+  /*
+   * Source URL or name of the accessory icon to show on the input.
+   * Shown while input has text in it. Not shown while input is used as a control component.
+   *
+   * @default - dismiss icon is shown.
+   * Set as null to disabled this icon.
+   */
+  accessoryIcon?: IconSourcesType | null;
+
+  /**
+   * A callback to call on accessoryIcon click event.
+   */
+  accessoryButtonOnPress?: (e: InteractionEvent) => void;
+
+  /**
+   * Label to display on top of the input.
+   */
+  label?: string;
+
+  /**
+   * Assistive text to display on the bottom of the input.
+   */
+  assistiveText?: string;
+
+  /**
+   * Secondary text to display on the right of the input.
+   */
+  secondaryText?: string;
+
+  /**
+   * Placeholder to display in the input.
+   */
+  placeholder?: string;
+
+  /**
+   * These props facilitate behaviour changes of the text field.
+   */
+  textInputProps?: Omit<TextInputProps, 'defaultValue' | 'value' | 'keyboardType' | 'onChangeText' | 'onBlur' | 'onFocus' | 'placeholder'>;
+
+  /**
+   * Optional string that triggers error state when non-empty.
+   * Error message replaces the assistive text and is displayed on the bottom of the input.
+   */
+  error?: string;
+
+  /**
+   * The value to show for the text input.
+   * Turns it into a controlled component.
+   * To prevent editting, use textInputProps and set editable as false.
+   */
+  value?: string;
+
+  /**
+   * Provides an initial value that will change when the user starts typing.
+   */
+  defaultValue?: string;
+
+  /**
+   * Determines which keyboard to open.
+   * enum("default", 'numeric', 'email-address', "ascii-capable", 'numbers-and-punctuation', 'url', 'number-pad', 'phone-pad', 'name-phone-pad',
+   * 'decimal-pad', 'twitter', 'web-search', 'visible-password')
+   * Determines which keyboard to open, e.g.numeric.
+   * The following values work across platforms: - default - numeric - email-address - phone-pad
+   * The following values work on iOS: - ascii-capable - numbers-and-punctuation - url - number-pad - name-phone-pad - decimal-pad - twitter - web-search
+   * The following values work on Android: - visible-password
+   */
+  type?: KeyboardTypeOptions | undefined;
```

Comment:
Can you help understand why you did not re-use [KeyBoardType](Callback that is called when the text input's submit button is pressed.

) for this

Reply:
Keyboardtype is used under the hood, exposed as "type" to match fluent web props, added same as comment